src/sortedDict.jl

```diff
@@ -399,7 +436,10 @@ function in{K,D,Ord <: Ordering}(pr::(@compat Tuple{Any,Any}), m::SortedDict{K,D
     return exactfound && isequal(m.bt.data[i].d,convert(D,pr[2]))
 end
 
-eltype{K,D,Ord <: Ordering}(m::SortedDict{K,D,Ord}) = (K,D)
+eltype{K,D,Ord <: Ordering}(m::SortedDict{K,D,Ord}) = @compat Tuple{K,D}
+eltype{K,D,Ord <: Ordering}(::Type{SortedDict{K,D,Ord}}) = @compat Tuple{K,D}
+keytype{K,D,Ord <: Ordering}(::SortedDict{K,D,Ord}) = K
+datatype{K,D,Ord <: Ordering}(::SortedDict{K,D,Ord}) = D
```

Comment:
To preserve the meaning, only these (except the `::Type` one) can be simplified.